homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -0,0 +1,108 @@
+"""Platform for sensor integration."""
+from __future__ import annotations
+
+from dataclasses import asdict
+import logging
+
+from homeassistant.components.sensor import (
+    ATTR_STATE_CLASS,
+    RestoreSensor,
+    SensorDeviceClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_DEVICE_CLASS, ATTR_UNIT_OF_MEASUREMENT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import dt as dt_util
+
+from . import DOMAIN
+from .const import GJ_TO_MWH, HEAT_METER_SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the sensor platform."""
+    _LOGGER.info("The Landis+Gyr Heat Meter sensor platform is being set up!")
+
+    unique_id = entry.entry_id
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    model = entry.data["model"]
+
+    device = DeviceInfo(
+        identifiers={(DOMAIN, unique_id)},
+        manufacturer="Landis & Gyr",
+        model=model,
+        name="Landis+Gyr Heat Meter",
+    )
+
+    sensors = []
+
+    for description in HEAT_METER_SENSOR_TYPES:
+        sensors.append(HeatMeterSensor(coordinator, unique_id, description, device))
+
+    async_add_entities(sensors)
+
+
+class HeatMeterSensor(CoordinatorEntity, RestoreSensor):
+    """Representation of a Sensor."""
+
+    def __init__(self, coordinator, unique_id, description, device):
+        """Set up the sensor with the initial values."""
+        super().__init__(coordinator)
+        self.key = description.key
+        self._attr_unique_id = f"{DOMAIN}_{unique_id}_{description.key}"
```

Comment:
We don't need to prefix the unique_id with the domain. The entity registry is aware of both the integration and the platform domain.

Reply:
Thanks. Could you help me with a way to change this without breaking backwards compatibility on the sensors that are already gathering statistics?

Reply:
We'd need to migrate the unique_id.

Here's a helper:
https://github.com/home-assistant/core/blob/43091a98562de0a4ccb689a034ad693f92a4c129/homeassistant/helpers/entity_registry.py#L1019-L1023

Example:
https://github.com/home-assistant/core/blob/43091a98562de0a4ccb689a034ad693f92a4c129/homeassistant/components/bmw_connected_drive/__init__.py#L73-L117